helion/autotuner/config_spec.py

```diff
@@ -131,11 +137,32 @@ def normalize(self, config: helion.Config | dict[str, object]) -> None:
             ("range_num_stages", self.range_num_stages, True),
             ("range_multi_buffers", self.range_multi_buffers, True),
             ("range_flattens", self.range_flattens, True),
+            ("static_ranges", self.static_ranges, True),
         ]:
             config[name] = mapping._normalize(
                 name, config.get(name, ()), flatten=flatten
             )
 
+        for block_id in self.static_ranges.valid_block_ids():
+            use_static_range = self.static_ranges.config_get(
+                config.get("static_ranges", ()),  # pyre-ignore[6]
+                block_id,
+            )
+
+            if use_static_range:
+                for name, mapping in (
+                    ("range_unroll_factors", self.range_unroll_factors),
+                    ("range_warp_specializes", self.range_warp_specialize),
+                    ("range_num_stages", self.range_num_stages),
+                    ("range_multi_buffers", self.range_multi_buffers),
+                    ("range_flattens", self.range_flattens),
+                ):
+                    if config[name]:  # The config is non empty
+                        # pyre-ignore[16]
+                        config[name][mapping.block_id_to_index(block_id)] = (
```

Comment:
The block_id may not exist in every one of these, since you can disable choices via the config_spec API.  Handle the case of it missing.

Maybe add a `mapping.set_config_to_default(config[name], block_id)` helper for this.